src/TensorTwoD.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include "config.h"
+#include "TensorTwoD.hpp"
+
+#include <cmath>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+
+#include "geopm/Exception.hpp"
+#include "TensorOneD.hpp"
+
+namespace geopm
+{
+
+
+    TensorTwoD::TensorTwoD(std::size_t rows, std::size_t cols)
+    {
+        set_dim(rows, cols);
+    }
+
+    TensorTwoD::TensorTwoD(const TensorTwoD &other)
+        : m_mat(std::move(other.m_mat))
```

Comment:
Wait! This is the copy constructor; don't attempt implementing move semantics here.

Reply:
Oh, thank you! Fixed.